runtime/jcl/common/reflecthelp.c

```diff
@@ -1952,3 +1952,72 @@ getRecordComponentsHelper(JNIEnv *env, jobject cls)
 	vmFuncs->internalExitVMToJNI(vmThread);
 	return result;
 }
+
+/* Create an array of string names for class's PermittedSubclasses */
+jarray
+permittedSubclassesHelper(JNIEnv *env, jobject cls)
+{
+	J9VMThread *vmThread = (J9VMThread *)env;
+	J9JavaVM *vm = vmThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	J9MemoryManagerFunctions *mmFuncs = vm->memoryManagerFunctions;
+	J9Class *clazz = NULL;
+	J9ROMClass *romClass = NULL;
+	jarray result = NULL;
+	J9Class *stringClass = NULL;
+	J9Class *stringArrayClass = NULL;
+	U_32 *permittedSubclassesCountPtr = 0;
+	j9array_t stringArrayObject = NULL;
+
+	vmFuncs->internalEnterVMFromJNI(vmThread);
+
+	clazz = J9VM_J9CLASS_FROM_JCLASS(vmThread, (jclass)cls);
+	romClass = clazz->romClass;
+
+	/* allocate String array for results */
+	stringClass = J9VMJAVALANGSTRING(vm);
+	if (NULL != vmThread->currentException) {
+		goto done;
+	}
+	stringArrayClass = fetchArrayClass(vmThread, stringClass);
+	if (NULL != vmThread->currentException) {
+		goto done;
+	}
+
+	permittedSubclassesCountPtr = getNumberOfPermittedSubclassesPtr(romClass);
+
+	stringArrayObject = (j9array_t) mmFuncs->J9AllocateIndexableObject(vmThread, stringArrayClass,
+				*permittedSubclassesCountPtr, J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
+	if (NULL == stringArrayObject) {
+		goto heapoutofmemory;
+	}
+
+	result = vmFuncs->j9jni_createLocalRef(env, (j9object_t)stringArrayObject);
```

Comment:
This only returns NULL if passed NULL as the object, so I don't think you need to native OOM handling at all.